tsdb/tsm1/engine.go

```diff
@@ -1104,15 +1116,66 @@ func (e *Engine) compactFull(ctx context.Context, grp CompactionGroup, wg *sync.
 
 	// Try the lo priority limiter, otherwise steal a little from the high priority if we can.
 	if e.compactionLimiter.TryTake() {
+		// Attempt to get ownership of the semaphore for this engine. If the
+		// default semaphore is in use then ownership will always be granted.
+		ttl := influxdb.DefaultLeaseTTL
+		lastCompaction := time.Duration(atomic.LoadInt64(&e.lastFullCompactionDuration))
+		if lastCompaction > ttl {
+			ttl = lastCompaction // If the last full compaction took > default ttl then set a new TTL
```

Comment:
What do you think about logging at Info that compaction is using a non-default value?

Reply:
That would be the expected case, and might lead to log messages every second whilst the engine was blocked waiting for ownership to the semaphore. We could add a log line when we get the semaphore though and it's not the `no-op` one?